### DISCUSSION TITLE
Running multiple instances of Authelia : Redis too ?

### DISCUSSION THREAD

**Discussion by Yericus** [authelia/authelia#3997] [Q&A] [answered]
Hi,

I very much began using Authelia for a week. please bear with me if this is obvious !
I want to protect multiple top domain names with authelia for my home setup.

I understand I need to run multiple instances of Authelia.
I use Docker desktop and Compose to create my containers.

My question : is one instance of Redis enough for multiple instances of Authelia in a same container in the same docker-compose.yml ?
Or should I create as many containers containing Redis and Authelia as needed (does Authelia always need internal port 9091 ?) ?

I am not sure how Authelia and Redis communicate and did not see any setting about that in server or redis pages for Authelia.
Thanks for any help !

**Comment by james-d-elliott** [comment, marked as answer]
Technically it should be fine. The session ID's are cryptographically randomly generated and should never overlap. We don't specifically recommend this, and instead recommend you utilize the database index option: https://www.authelia.com/configuration/session/redis/#database_index which segments all sessions between the indexes.

**Comment by Yericus** [comment]
Allright time to experiment then !
Thank you :)

**Reply by Yericus** [reply]
Finally got to set this up today and it's working fine.
I copied folder and files from my first authelia setup and updated port numbers and domains + session random # for each config file of each instance of authelia.

Added 2 other authelia services on different ports and with a different container name in my yml docker compose file and it just ran as is.
Copying the folders also meant the 2fa was the same for all which I wanted to.

This is for a non critical home setup so until I go further in LDAP and database index, this works well enough !
